container/announcement.go

```diff
@@ -2,76 +2,97 @@ package container
 
 import (
 	"github.com/nspcc-dev/neofs-api-go/v2/container"
+	"github.com/nspcc-dev/neofs-api-go/v2/refs"
 	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
 )
 
 // UsedSpaceAnnouncement is an announcement message used by storage nodes to
 // estimate actual container sizes.
+//
+// UsedSpaceAnnouncement is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/container.UsedSpaceAnnouncement
+// message. See ReadFromV2 / WriteToV2 methods.
+//
+// Instances can be created using built-in var declaration.
+//
+// Note that direct typecast is not safe and may result in loss of compatibility:
+// 	_ = UsedSpaceAnnouncement(accounting.UsedSpaceAnnouncement{}) // not recommended
 type UsedSpaceAnnouncement container.UsedSpaceAnnouncement
 
-// NewAnnouncement initialize empty UsedSpaceAnnouncement message.
+// ReadFromV2 reads UsedSpaceAnnouncement from the container.UsedSpaceAnnouncement message.
 //
-// Defaults:
-//  - epoch: 0;
-//  - usedSpace: 0;
-//  - cid: nil.
-func NewAnnouncement() *UsedSpaceAnnouncement {
-	return NewAnnouncementFromV2(new(container.UsedSpaceAnnouncement))
+// See also WriteToV2.
+func (a *UsedSpaceAnnouncement) ReadFromV2(m container.UsedSpaceAnnouncement) {
+	*a = UsedSpaceAnnouncement(m)
 }
 
-// NewAnnouncementFromV2 wraps protocol dependent version of
-// UsedSpaceAnnouncement message.
+// WriteToV2 writes UsedSpaceAnnouncement to the container.UsedSpaceAnnouncement message.
+// The message must not be nil.
 //
-// Nil container.UsedSpaceAnnouncement converts to nil.
-func NewAnnouncementFromV2(v *container.UsedSpaceAnnouncement) *UsedSpaceAnnouncement {
-	return (*UsedSpaceAnnouncement)(v)
+// See also ReadFromV2.
+func (a UsedSpaceAnnouncement) WriteToV2(m *container.UsedSpaceAnnouncement) {
+	*m = (container.UsedSpaceAnnouncement)(a)
 }
 
 // Epoch of the announcement.
```

Comment:
I propose to not consider method name as a word:
```suggestion
// Epoch returns NeoFS epoch when UsedSpaceAnnouncement was made.
```
See other methods too.